aten/src/ATen/native/mps/operations/Pad.mm

```diff
@@ -107,23 +107,20 @@
     grad_output = grad_output_.contiguous();
   }
 
-  const int64_t input_dim = input.dim();
-  MPSShape *leftPadding = nullptr, *rightPadding = nullptr;
-  if (padding_dim == 3) {
-    leftPadding = [NSArray arrayWithObjects:(const NSNumber*[]){ @(0), @(0), @(pad_front), @(pad_t), @(pad_l) } count:input_dim];
-    rightPadding = [NSArray arrayWithObjects:(const NSNumber*[]){ @(0), @(0), @(pad_back), @(pad_b), @(pad_r) } count:input_dim];
-  } else if (padding_dim == 2) {
-    leftPadding = [NSArray arrayWithObjects:(const NSNumber*[]){ @(0), @(0), @(pad_t), @(pad_l) } count:input_dim];
-    rightPadding = [NSArray arrayWithObjects:(const NSNumber*[]){ @(0), @(0), @(pad_b), @(pad_r) } count:input_dim];
-  } else if (padding_dim == 1) {
-    if (input_dim > 1) {
-      leftPadding = [NSArray arrayWithObjects:(const NSNumber*[]){ @(0), @(0), @(pad_l) } count:input_dim];
-      rightPadding = [NSArray arrayWithObjects:(const NSNumber*[]){ @(0), @(0), @(pad_r) } count:input_dim];
-    } else {
-      leftPadding = [NSArray arrayWithObjects:(const NSNumber*[]){ @(pad_l) } count:input_dim];
-      rightPadding = [NSArray arrayWithObjects:(const NSNumber*[]){ @(pad_r) } count:input_dim];
-    }
+  std::vector<NSNumber*> leftPadVec(ndims, @(0));
+  std::vector<NSNumber*> rightPadVec(ndims, @(0));
+  leftPadVec [ndims - 1] = @(pad_l);
```

Comment:
Could a scalar be passed to this function and `ndim` to be 0?

Reply:
No, that should be caught with this error :
TypeError: pad(): argument 'input' (position 1) must be Tensor, not int